src/App.tsx

```diff
@@ -1,155 +1,124 @@
-
-            <div className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          {/* This todo is in loadind state */}
-          <div className="todo">
-            <label className="todo__status-label">
-              <input type="checkbox" className="todo__status" />
-            </label>
-
-            <span className="todo__title">Todo is being saved now</span>
-            <button type="button" className="todo__remove">×</button>
-
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
-
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer">
-          <span className="todo-count">
-            3 items left
-          </span>
-
-          {/* Active filter should have a 'selected' class */}
-          <nav className="filter">
-            <a href="#/" className="filter__link selected">
-              All
-            </a>
-
-            <a href="#/active" className="filter__link">
-              Active
-            </a>
-
-            <a href="#/completed" className="filter__link">
-              Completed
-            </a>
-          </nav>
-
-          {/* don't show this button if there are no completed todos */}
-          <button type="button" className="todoapp__clear-completed">
-            Clear completed
-          </button>
-        </footer>
+        <TodoAppHeader
+          todos={todos}
+          addTodo={addTodo}
+          newTitle={newTitle}
+          setNewTitle={setNewTitle}
+        />
+
+        <TodoList todos={todoList} />
+
+        {todos.length ? (
+          <TodoFooter
+            todos={todos}
+            filterType={filterType}
+            setFilterType={setFilterType}
+          />
+        ) : null}
       </div>
 
-      {/* Notification is shown in case of any error */}
-      {/* Add the 'hidden' class to hide the message smoothly */}
-      <div className="notification is-danger is-light has-text-weight-normal">
+      <div
+        className={classNames(
+          'notification',
+          'is-danger',
+          'is-light',
+          'has-text-weight-normal',
+          {
+            hidden: !hasTitleError,
+          },
+        )}
+      >
         <button type="button" className="delete" />
 
-        {/* show only one message at a time */}
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
+        {newTitle.trim().length === 0 && (
+          'Title can`t be empty'
+        )}
+
+        {newTitle.trim().length !== 0 && (
+          'Unable to add a todo'
+        )}
         <br />
-        Unable to update a todo
       </div>
```

Comment:
create separate `Notification` component and create different error types by using `enum Error`